services/settings-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/SettingsServiceDAL.java

```diff
@@ -164,6 +172,20 @@ public List<Map<String, Object>> rdsOptions() {
         return orderableOptionsByRegion;
     }
 
+    public List<CertificateSummary> acmCertificateOptions() {
+        try {
+            // only list certificates that aren't expired, invalid, revoked, or otherwise unusable
+            ListCertificatesResponse response = acm.listCertificates(ListCertificatesRequest.builder()
+                    .certificateStatuses(List.of(CertificateStatus.PENDING_VALIDATION, CertificateStatus.ISSUED))
+                    .build());
+            LOGGER.info("ACM PENDING_VALIDATION and ISSUED certs: {}", response);
+            return response.certificateSummaryList();
```

Comment:
We return `List.of()` in our catch, but we don't check for `response.hasCertificateSummaryList()` in the try so we theoretically could return a `null`

Reply:
Good callout, but it turns out that [`ListCertificatesResponse#hasCertificateSummaryList`](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/acm/model/ListCertificatesResponse.html#hasCertificateSummaryList--) "is useful because the SDK will never return a null collection or map, but you may need to differentiate between the service returning nothing (or null) and the service returning an empty collection or map".

In this case we don't need to differentiate: the service responding with no certificate summary list (but also no exception) is functionally the same as returning an empty list. Plus, documentation says the SDK will never return a null collection/map.

But, I agree that we should `null` protect, just in case. So I'll implement that in the next diff.